pulser-core/pulser/_hamiltonian_data/hamiltonian_data.py

```diff
@@ -725,62 +745,119 @@ def _finite_waist_amp_fraction(
         # by the propagation direction), ie
         return float(np.exp(-((dist / laser_waist) ** 2)))
 
-    def _create_noise_trajectory(self) -> None:
+    def _create_noise_trajectories(self) -> None:
         """Updates noise random parameters.
 
         Used at the start of each run. If SPAM isn't in chosen noises, all
         atoms are set to be correctly prepared.
         """
+        self.noise_trajectories: list[tuple[NoiseTrajectory, int]] = []
```

Comment:
Also, the meaning for the `int` in the `tuple` is quite mysterious, especially given that `noise_trajectories` is a public attribute (should it be, actually?).
I don't remember the reason why it couldn't be added to the `NoiseTrajectory` directly but I remember there was one :sweat_smile: 
I think a nice compact way to make its meaning clear would be to used a `NamedTuple` here, what do you think?